### ISSUE TITLE
Js generation doesn't work for primitive slices

### ISSUE THREAD

**Issue by jcrist1** [rust-diplomat/diplomat#699]
When trying to generate some js bindings for a library I was working on, I ran into an issue with a slice return method. I tried to have look through the existing issues but couldn't find anything so I created a minimal example here: https://github.com/jcrist1/diplomat-dbg/tree/6649bacd8e004f82814510774acde572625be891/js_slice

The basic problem is that the generated method for a static method like:
```rs
#[diplomat::bridge]
pub mod ffi {
    const DATA: &[u8] = &[1, 2, 3, 4, 5, 6];

    #[allow(unused)]
    #[diplomat::opaque]
    pub struct SliceGetter();

    impl SliceGetter {
        pub fn get_slice<'a>() -> &'a [u8] {
            DATA
        }
    }
}
```
generates the following static method
```js
    static getSlice() {
        const diplomatReceive = new diplomatRuntime.DiplomatReceiveBuf(wasm, 8, 4, false);
        
        // This lifetime edge depends on lifetimes 'a
        let aEdges = [];
        
        const result = wasm.SliceGetter_get_slice(diplomatReceive.buffer);
    
        try {
            return new diplomatRuntime.DiplomatSlicePrimitive.getSlice(wasm, diplomatReceive.buffer, "u8", aEdges);
        }
        
        finally {
            diplomatReceive.free();
        }
    }
```

Where the line
```js
diplomatRuntime.DiplomatSlicePrimitive.getSlice
```
is not a constructor.
If we change it to
```js
            return new diplomatRuntime.DiplomatSlicePrimitive(wasm, diplomatReceive.buffer, "u8", aEdges).buffer;
```
Then we are able to correctly call the method.

**Comment by Manishearth** [comment]
Yeah the JS slice stuff has been buggy for a while, we should just fix this

**Comment by Manishearth** [comment]
I think we should still call getSlice, the buffer is always a u8 slice I think. We should just add parens for it.

**Comment by jcrist1** [comment]
> I think we should still call getSlice, the buffer is always a u8 slice I think. We should just add parens for it.

I'm a bit confused by this. I can't find a `getSlice` method on `DiplomatSlicePrimitive` do you mean `getValue`?

**Comment by Manishearth** [comment]
yeah, sorry, and it seems like `.buffer` is already typed, so that should be fine.

**Comment by jcrist1** [comment]
Fixed in #722
